app.py

```diff
@@ -44,11 +49,13 @@ def require_login(user_param_name='user'):
     def inner_decorator(func):
         @wraps(func)
         def wrapped_func(*args, **kwargs):
-            token = None
-            if request.method == 'GET':
-                token = request.args.get('token')
-            elif request.method == 'POST':
-                token = request.values.get('token')
+            token = request.args.get('token')
+            if token is not None:
+                if token != 'undefined':
```

Comment:
Why this check? It seems unnecessary

Reply:
Otherwise it decides to bounce - The cookie is set to None, undefined, <token>, and finally undefined - thus making the site unusable. I think it has something to do with an empty `?token=` GET parameter?

Reply:
With code
```python
token = args.get(...)
print(token)
if token is not None
    >
```
The following is output for `?token=123`
```
123

undefined
None
```
With no GET and undefined as the value of the cookie:
```
None
<token>
None

undefined
None
```
And with a valid token as cookie value (worst of all):
```
None

undefined
None
```
Absolutely unusable, in other words. The final value of the cookie is always undefined, which is not a valid token.

With code
```python
token = args.get(...)
print(token)
if token != 'undefined`
    >
```
The result is Internal Server Error as it tries to set a None cookie.

Reply:
With both if statements, it gives the expected behaviour of setting the cookie to the value of `?token=<val>`. If a user puts an invalid one there it puts a correct one upon reloading.

Reply:
It seems that the frontend requests `/me?token=...` on every page load and it is trying to get the token from the now missing GET parameter. That is why we get requests like this `/me?token=undefined`.

Look in `/src/App.js`. You can remove the part where the code manually takes the token from the URL and sends it as a GET parameter. Since we are using cookies now all this will happen in the background. When we make a request to `/me` the cookie will automatically be passed along by the browser.

Reply:
@brynolv Will this be resolved?

Reply:
When/(if) I find the time? A strong "Maybe" right now.

Reply:
Fixed in d8a8364, however 24c9803 appears to have problems...

Reply:
Want to leave it at this? I am no where close to good enough with JS/Node to guess why the searchbar disappears with `npm audit fix`.
@DokID @mattiasgrenfeldt